utils/lru.go

```diff
@@ -131,6 +132,32 @@ func (c *Cache) Get(key interface{}) (value interface{}, ok bool) {
 	return nil, false
 }
 
+// GetOrAdd returns the existing value for the key if present. Otherwise, it stores and returns the given value. The loaded result is true if the value was loaded, false if stored.
+func (c *Cache) GetOrAdd(key, value interface{}) (actual interface{}, loaded bool) {
+	return c.GetOrAddWithExpiresInSecs(key, value, 0)
+}
+
+// GetOrAddWithDefaultExpires is a variant of GetOrAdd that uses the default expiry configured for the cache.
+func (c *Cache) GetOrAddWithDefaultExpires(key, value interface{}) (actual interface{}, loaded bool) {
+	return c.GetOrAddWithExpiresInSecs(key, value, c.defaultExpiry)
+}
+
+// GetOrAddWithExpiresInSecs is a variant of GetOrAdd that uses the given expiry configured for the cache.
+func (c *Cache) GetOrAddWithExpiresInSecs(key interface{}, value interface{}, expireAtSecs int64) (actual interface{}, loaded bool) {
```

Comment:
I modelled this interface after `sync.Map`'s functionality. I originally thought I'd have to swap out the implementation to actually use a `sync.Map`, but this whole data structure is locked anyway, so there's no race conditions.

We may want to consider using `sync.Map` in the future for a performance boost, given the likely semantics of most of our caches.